### PR TITLE
#21: EXPERIMENTAL: Field Conversion Improvements

### PR DESCRIPTION
**Pull request by nvamelichev**
**This PR brings an experimental feature**, explicitly annotated as `@ExperimentalApi`:

Adding arbitrary converters between Java values and DB-compatible "column values", using new `@CustomValueType` annotation either on the type, or inside `@Column` annotation on the entity field. 

These "custom value types" can even be used inside IDs, if their converters transform field value into a ID-compatible value (a String, a enum, an integer, a timestamp, a boolean value or a byte array).

This feature has the potential to replace all ad-hoc "string-value type" functionality in the future.

### CODE REVIEW

File: repository-test/src/main/java/tech/ydb/yoj/repository/test/sample/model/Snapshottable.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import tech.ydb.yoj.repository.db.Entity;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.UUID;
+
+import static lombok.AccessLevel.PRIVATE;
+
+@Value
+public class Snapshottable implements Entity<Snapshottable> {
```
Comment (lavrukov) [inline repository-test/src/main/java/tech/ydb/yoj/repository/test/sample/model/Snapshottable.java:17]: nit: SnapshotTable?

**Comment by nvamelichev** [comment]
**This is almost done**, just needs support of `@CustomValueType` and `@StringValueType` annotations on columns, or, maybe `customValueType` and `stringValueType` arguments for the `@Column` annotation. 

This will resolve the need to e.g. globally register `UUID` as a string-value type.

**Comment by nvamelichev** [comment]
**UPD:** This is done, and custom value converter actually works quite fine, including **in IDs**

File: databind/src/main/java/tech/ydb/yoj/databind/StringValueConverter.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
+package tech.ydb.yoj.databind;
```
Comment (lavrukov) [inline databind/src/main/java/tech/ydb/yoj/databind/StringValueConverter.java:1]: suggestion: maybe move it to separate package like tech.ydb.yoj.databind.converter ?

**Review by lavrukov — APPROVED** [review]

File: databind/src/main/java/tech/ydb/yoj/databind/StringValueConverter.java (hunk 2 above)
Comment (nvamelichev) [reply databind/src/main/java/tech/ydb/yoj/databind/StringValueConverter.java]: Good idea, moved `*Converter` classes to a separate package! :+1:

**Pull request merged**